Add JetBrains Rider (#30684)
dotnet/docs: docs/fundamentals/tools-and-productivity.md

## IDEs

> You can write .NET applications in [Visual Studio Code](https://code.visualstudio.com/docs), [Visual Studio](/visualstudio/windows), [Visual Studio for Mac](/visualstudio/mac), or [JetBrains Rider](https://www.jetbrains.com/rider/).

Reviewer: ```suggestion
You can write .NET applications in [Visual Studio Code](https://code.visualstudio.com/docs), [Visual Studio](/visualstudio/windows), [Visual Studio for Mac](/visualstudio/mac), or with [other .NET tools](https://dotnet.microsoft.com/platform/tools).
```

Since we do link to JetBrains on our marketing page, we could link to that page from here.
